src/System.Security.Cryptography.X509Certificates/tests/PfxTests.cs

```diff
@@ -28,16 +28,34 @@ public static IEnumerable<object[]> BrainpoolCurvesPfx
         [MemberData(nameof(StorageFlags))]
         public static void TestConstructor(X509KeyStorageFlags keyStorageFlags)
         {
-            byte[] expectedThumbprint = "71cb4e2b02738ad44f8b382c93bd17ba665f9914".HexToByteArray();
-
             using (var c = new X509Certificate2(TestData.PfxData, TestData.PfxDataPassword, keyStorageFlags))
             {
-                string subject = c.Subject;
-                Assert.Equal("CN=MyName", subject);
-                byte[] thumbPrint = c.GetCertHash();
-                Assert.Equal(expectedThumbprint, thumbPrint);
+                VerifyConstructor(c);
+            }
+        }
+
+#if netcoreapp11
+        [Theory]
+        [MemberData(nameof(StorageFlags))]
+        public static void TestConstructor_SecureString(X509KeyStorageFlags keyStorageFlags)
+        {
+            using (SecureString password = TestData.CreatePfxDataPasswordSecureString())
+            using (var c = new X509Certificate2(TestData.PfxData, password, keyStorageFlags))
+            {
+                VerifyConstructor(c);
             }
         }
+#endif
+
+        private static void VerifyConstructor(X509Certificate2 c)
```

Comment:
Please inline this.  The function only works if the correct data was cracked, and that's embedded in the caller.  There are way too many different test certs for this to be reasonable.

Alternatively, a very specific name that indicates that it'll only work with the correct data blob being processed by the caller.